pkg/rpcserver/rpcserver.go

```diff
@@ -306,25 +328,23 @@ func (serv *server) handleRunnerConn(ctx context.Context, eg *errgroup.Group,
 		return err
 	}
 
-	serv.checkOnce.Do(func() {
-		// Run the machine check.
-		eg.Go(func() error {
-			if err := serv.runCheck(ctx, &info); err != nil {
-				return fmt.Errorf("%w: %w", errFatal, err)
-			}
-			return nil
-		})
-	})
-
-	if serv.triagedCorpus.Load() {
-		eg.Go(runner.SendCorpusTriaged)
+	select {
+	case <-ctx.Done():
+	case onHandshake <- &info:
 	}
 
+	ctx2, cancel := context.WithCancel(ctx)
+	eg, ctx3 := errgroup.WithContext(ctx2)
+	eg.Go(runner.SendCorpusTriaged)
+	eg.Go(func() error {
+		defer cancel()
```

Comment:
What is this cancelling?

Reply:
In this particular case, we have
1) `runner.SendCorpusTriaged` that, if fails, should abort everything. That's achieved via the errgoup and errgroup's context.
2) `serv.connectionLoop` that could only be forcefully aborted by calling `runner.Stop()`.

If I make `connectionLoop` take a context object and react to its cancellation itself, it should become more readable -- at least there won't be `context.WithCancel` within the `handleRunnerConn` function.